data/versions.yml

```diff
@@ -1,5 +1,26 @@
-# The version of Istio currently documented in istio.io
 main: "1.26"
 
-# The version of Istio currently documented in preliminary.istio.io
 preliminary: "1.27"
+
+# List of all versions to be included in the dropdown selector and for archive logic
+versions:
```

Comment:
I think you can infer all of this programatically.

We know what the version is, we can subtract N from it, and when the site is built, it's built with a flag showing if content is archive, preliminary or current.  (At least that used to be true; we can just infer anything that is not latest is archive.)